src/helpers.js

```diff
@@ -87,18 +93,34 @@ export function applySugar(environment: Object) {
 }
 
 export function getCommand(): { command: string, options: Object, parameters: Array<string> } {
+  const pythonFile = Path.join(Path.dirname(__dirname), 'index.py')
+  const shellScript = coolTrim`
+    if [ "$(which python)" != "" ]; then
+      python ${pythonFile}
+    elif [ "$(which python3)" != "" ]; then
+      python3 ${pythonFile}
+    elif [ "$(which python2.7)" != "" ]; then
+      python2.7 ${pythonFile}
+    else
+      env
+    fi
+  `
   const command = process.env.SHELL || 'sh'
-  const options = { encoding: 'utf8' }
-  let parameters = ['-c', 'env;exit']
+  const options: Object = { encoding: 'utf8' }
+  const stdin = []
 
   const shell = Path.basename(command)
   if (shell === 'bash') {
-    parameters = ['-c', 'source ~/.bashrc;source ~/.bash_profile;env;exit']
+    stdin.push('source ~/.bashrc 2>/dev/null')
+    stdin.push('source ~/.bash_profile 2>/dev/null')
   } else if (shell === 'zsh') {
-    parameters = ['-c', 'source ~/.zshrc;env;exit']
+    stdin.push('source ~/.zshrc 2>/dev/null')
   } else if (shell === 'fish') {
-    parameters = ['-c', 'source ~/.config/fish/config.fish;env;exit']
+    stdin.push('source ~/.config/fish/config.fish 2>/dev/null')
   }
+  stdin.push(shellScript)
```

Comment:
This doesn't work. The shell script is not compatible with many shells (like **fish** or **csh**).